src/docker.ts

```diff
@@ -36,6 +36,18 @@ const saveDockerImages = async (): Promise<void> => {
       `Cache miss occurred on the primary key ${key}. Not saving cache as ` +
         "read-only option was selected.",
     );
+    /* Check if a cache with our key has been saved between when we checked in
+     * loadDockerImages and now.
+     */
+  } else if (
+    key === (await restoreCache([""], key, [], { lookupOnly: true }))
+  ) {
+    info(
+      "A cache miss occurred during the initial attempt to load Docker " +
+        `images. Subsequently a cache with a matching key, ${key}, was saved. ` +
+        "This can occur when docker-cache is used by multiple jobs run in " +
+        "parallel. Not saving cache.",
```

Comment:
Nit: I have a tendency to limit self-references to the name of a tool because it makes it more tedious to change in the future (e.g., "This can occur when run in parallel.". That doesn't seem very likely to me here, but it's also a way to make this log statement a little less verbose. It might also help to join the first two sentences via a "but" to clarify their logical relationship.